libsupport/include/katana/EntityTypeManager.h

```diff
@@ -78,6 +83,181 @@ class EntityTypeManager {
         atomic_entity_type_id_to_entity_type_ids_(
             std::move(atomic_entity_type_id_to_entity_type_ids)) {}
 
+  template <typename ArrowType>
+  struct PropertyColumn {
+    int field_index;
+    std::shared_ptr<ArrowType> array;
+
+    PropertyColumn(int i, std::shared_ptr<ArrowType>& a)
+        : field_index(i), array(a) {}
+  };
+
+  /// This function can be used to convert "old style" graphs (storage format 1,
+  /// where types are represented by bool or uint8 properties) and "new style"
+  /// graphs (version > 2, where types are represented in our native type
+  /// represenation). This function is serial but it likely iterates over
+  /// O(nodes) and O(edges) vectors, so it is very slow. It should only be used
+  /// for updating old graphs.
+  ///
+  /// The number of entries in entity_type_ids should be equal to topo_size and
+  /// properties->num_rows().
+  template <template <typename> class Vector>
+  static Result<void> AssignEntityTypeIDsFromProperties(
```

Comment:
Why are you moving this to libsupport? It seems odd since libsupport is not really "property-aware" and just deals with low-level vectors and arrays for the most part.

Where else do you expect this to be used other than with PropertyGraph?

Reply:
I am not exactly sure what you mean here, but I will try to address your concerns. libsupport is now type-aware and due to our previous type representation, being type-aware requires a certain amount of property-awareness.

I don't actually need this function to be in libsupport, I just need it to be publicly accessible. Because one purpose of this function is to initialize an `EntityTypeManager`, I thought this would be a natural home for it. This seemed like it would be an awkward appendage to `PropertyGraph`.

The high level answer to your second question is that this function is required anywhere we need to update an old-style graph to a new-style graph. The specific answer is that I am working on a code path in the enterprise repo that loads graphs without creating `PropertyGraph`s.